azext_iot/tests/iothub/core/test_iothub_storage_int.py

```diff
@@ -149,51 +149,62 @@ def test_storage(self):
             checks=self.is_empty(),
         )
 
-        self.cmd(
-            'iot hub device-identity export -n {} --bcu "{}"'.format(
-                self.entity_name, self.live_storage_uri
-            ),
-            checks=[
-                self.check("outputBlobContainerUri", self.live_storage_uri),
-                self.check("failureReason", None),
-                self.check("type", "export"),
-                self.check("excludeKeysInExport", True),
-                self.exists("jobId"),
-            ],
-        )
+        attempts = 0
+        setup_completed = False
+        while not setup_completed:
+            try:
+                self.check_for_running_import_export()
```

Comment:
Why are we doing **check_for_running_import_export** only for this test and not for the other 2 tests as well?

Reply:
The other two tests have identity assignment followed by sleep, which should provide a time safeguard for ensuring that the job completes. The test_storage with the sas tokens does not have this, and thus it is needed if the while loop is retried